packages/expect/src/jestMatchersObject.ts

```diff
@@ -63,21 +63,21 @@ export const setMatchers = (
 
       class CustomMatcher extends AsymmetricMatcher<[unknown, unknown]> {
         constructor(inverse: boolean = false, ...sample: [unknown, unknown]) {
-          super(sample);
-          this.inverse = inverse;
+          super(sample, inverse);
         }
 
         asymmetricMatch(other: unknown) {
-          const {pass} = matcher(
+          const {pass} = matcher.call(
+            this.matcherState,
```

Comment:
here's the real fix - our own matchers "cheat" by mot using `this` but rather importing `equals` etc directly